### PR TITLE
Implemented `selectionFeedback` for ComboBox and Dropdown

### PR DESCRIPTION
**Pull request by Licen-it**
Closes IBM/carbon-components-angular#1362

This is implementing the `selectionFeedback` behaviour that you can find in the react version of ComboBox and Dropdown components

#### Changelog

**New**

* `selectionFeedback` property can now be configured for ComboBox
* `selectionFeedback` property can now be configured for Dropdown

P.S.:
I was also fixing the invalid icon misalignment in ComboBox, but pulling latest code i found someone fixed it as well.

### CODE REVIEW

File: src/dropdown/abstract-dropdown-view.class.ts [hunk 1]
```diff
@@ -99,4 +99,8 @@ export class AbstractDropdownView {
 	 * Subscribe the function passed to an internal observable that will resolve once the items are ready
 	 */
 	onItemsReady(subcription: () => void): void {}
+	/**
+	 * Reorder selected items bringing them to the top of the list
+	 */
+	reorderSelected(andMoveFocus?: boolean): void {}
```
Comment (Donisius) [inline src/dropdown/abstract-dropdown-view.class.ts:105]: maybe just
```suggestion
	reorderSelected(moveFocus?: boolean): void {}
```

File: src/dropdown/list/dropdown-list.component.ts [hunk 2]
```diff
@@ -519,4 +521,11 @@ export class DropdownList implements AbstractDropdownView, AfterViewInit, OnDest
 		// this subscription will auto unsubscribe because of the `first()` pipe
 		(this._itemsReady || of(true)).pipe(first()).subscribe(subcription);
 	}
+
+	reorderSelected(andMoveFocus = false): void {
```
Comment (Donisius) [inline src/dropdown/list/dropdown-list.component.ts:525]:
```suggestion
	reorderSelected(moveFocus = false): void {
```

File: src/dropdown/list/dropdown-list.component.ts [hunk 3]
```diff
@@ -519,4 +521,11 @@ export class DropdownList implements AbstractDropdownView, AfterViewInit, OnDest
...
+		this.displayItems = [...this.getSelected(), ...this.getListItems().filter(item => !item.selected)];
+		if (andMoveFocus) {
```
Comment (Donisius) [inline src/dropdown/list/dropdown-list.component.ts:527]:
```suggestion
		if (moveFocus) {
```

File: src/dropdown/list/dropdown-list.component.ts [hunk 4]
```diff
@@ -519,4 +521,11 @@ export class DropdownList implements AbstractDropdownView, AfterViewInit, OnDest
 		// this subscription will auto unsubscribe because of the `first()` pipe
 		(this._itemsReady || of(true)).pipe(first()).subscribe(subcription);
 	}
+
+	reorderSelected(moveFocus = false): void {
+		this.displayItems = [...this.getSelected(), ...this.getListItems().filter(item => !item.selected)];
+		if (moveFocus) {
+			setTimeout(() => this.getCurrentElement().focus(), 1);
```
Comment (cal-smith) [inline src/dropdown/list/dropdown-list.component.ts:528]: We should be able to elide the explicit timeout and have it fire on the next tick ... unless this is to account for focus delays or something?

```suggestion
			setTimeout(() => this.getCurrentElement().focus());
```

**Pull request merged**